libgalois/src/analytics/subgraph_extraction/subgraph_extraction.cpp

```diff
@@ -49,16 +88,18 @@ SubGraphNodeSet(
       katana::iterate(uint32_t(0), uint32_t(num_nodes)),
       [&](const uint32_t& n) {
         uint32_t src = node_set[n];
-        auto edge_range = graph->topology().edge_range(src);
+
+        auto first = graph->edges(src).begin();
+        auto last = graph->edges(src).end();
         for (auto dest : node_set) {
           // Binary search on the edges sorted by destination id
-          auto edge_matched = std::lower_bound(
-              edge_iterator(edge_range.first), edge_iterator(edge_range.second),
-              dest);
-          if (*edge_matched != edge_range.second) {
-            while (*edge_matched == dest) {
+          auto lower_bound = LowerBound(
+              first, last, LessThan<katana::PropertyGraph>(*graph, dest));
+          if (lower_bound != last) {
```

Comment:
Could you use `FindEdgeSortedByDest` here?
https://github.com/KatanaGraph/katana/blob/bc9a6c665bc8aeb1920c3ebb8567a6563c5162d3/libgalois/include/katana/TypedPropertyGraph.h#L193-L205

Reply:
If we can use `FindEdgeSortedByDest` we may not need `LowerBound` at all in this algorithm.